sky/backends/backend_utils.py

```diff
@@ -1057,6 +1057,16 @@ def get_node_ips(cluster_yaml: str,
                  head_ip_max_attempts: int = 1,
                  worker_ip_max_attempts: int = 1) -> List[str]:
     """Returns the IPs of all nodes in the cluster."""
+
+    # When ray up launches TPU VM Pod, Pod workers (except for the head)
+    # won't be connected to Ray cluster. Thus "ray get-worker-ips"
+    # won't work and we need to query the node IPs with gcloud as
+    # implmented in _get_tpu_pod_ips.
+    ray_config = common_utils.read_yaml(cluster_yaml)
+    use_tpu_vm = ray_config['provider'].get('_has_tpus', False)
+    if use_tpu_vm:
```

Comment:
Don't we have to check `is_tpu_pod` here?

Reply:
Yeah great catch.. The tricky problem here is checking `is_tpu_pod` is not as easy as checking `is_tpu_vm` as when we are only given the Ray YAML `cluster_yaml` but not the handle we created. 

To check whether it's a tpu pod or not for Ray YAML we need to check whether
`config['available_node_types']['ray_head_default']['node_config']['acceleratorType'] is in ['v2-32', ...]`.
not the cleanest code I think.

As `gcloud` works for single device and pods TPU VM so I'd slightly prefer keeping this. wdty?

Reply:
OK. Then I think it's fine. Thanks for the explanation.